### PR TITLE
Set Availability Use Case Implementation

### PR DESCRIPTION
**Pull request by od-obas1187**
I just completed my implementation of the Set Availability Use Case package!

Compared to my previous closed PR - which was an EXTREMELY rough implementation of the SetAvailability use case, essentially just to give the UI people an idea about what to pass to the setAvailability controller - this is the full thing, which should now adhere properly to the Clean Architecture guidelines.

It essentially goes through the following User Story: the AvailabilityController is called witha list of LocalTime values representing when the user is available. This controller then calls the AvailabilityInputBoundary, which is implemented by AvailablityInteractor, and has a method named create() that adds TimeBlocks to the user's intially empty TimeBlockManager. After this is completed, the AvailabiltyInteractor passes an AvailabilityModel back to the AvailabityOutputBoundary, an interface implemented by the AvailablityPresenter, and the presenter then passes the updated AvailabilityModel to the view model, which can then be unpackaged for the dashboard to then present the information to the user through the UI. I also made sure the availability is assigned to a schedule associated with the curriculum, thus linking the availability to the user.

My next PR will be to implement tests (should be fun :) )

Would appreciate some pointers!

### CODE REVIEW

File: src/main/java/set_availability/SetAvailabilityController.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+        this.userAvailabilityInteractor = interactor;
+
+    }
+
+    /**
+     * 1. Receives set of String values (user’s availabilities);
+     * 2. Sends them to the SetAvailabilityInteractor
+     */
+    public void create(String[] availabilities, int curriculumId) {
```
Comment (MMachadoUofT) [inline src/main/java/set_availability/SetAvailabilityController.java:28]: Good idea using a primitive array here! It'll definitely make passing information down into the use case layer much easier.

File: src/main/java/set_availability/SetAvailabilityUseCase.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+     * 2. Using that, create a CommonTimeBlockManager factory that further creates a CommonTimeBlockManager
+     * 3. Create a SetAvailability Presenter
+     * 4. Return that presenter
+     *
+     * @param availabilityInputs an array...
+     */
+    @Override
+    public void create(String[] availabilityInputs, int curriculumId) {
+
+        //Create TimeBlockManager
```
Comment (MMachadoUofT) [inline src/main/java/set_availability/SetAvailabilityUseCase.java:31]: We wanna make sure that we don't have redundant code; it's actually a step in the process of refactoring. In fact, this comment is a little misleading, as we aren't creating a TimeBlockManager, rather we are just setting the TimeBlocks for the existing one.

File: src/main/java/set_availability/SetAvailabilityUseCase.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        Schedule schedule = InMemoryUser.getActiveUser().getSchedule();
+        TimeBlockManager timeBlockManager = schedule.getAvailability();
+        timeBlockManager.getTimeBlocks().forEach(
```
Comment (MMachadoUofT) [inline src/main/java/set_availability/SetAvailabilityUseCase.java:34]: This part is actually a bit redundant, as well; I thought I had mentioned this, but I'm having the TimeBlockManager class implement the Iterable interface, so you'll just be able to do `for (TimeBlock t : TimeBlockManager) {}`

**Review by MMachadoUofT — APPROVED** [review]
This looks pretty solid! The only thing I would want changed would be to have your interaction loop over the Manager directly, though since that hasn't been implemented yet, I can see why you can't do this. I'll be adding this to the next PR I open, so you'll be able to change this relatively soon. Good work!

**Review by Amir1441 — APPROVED** [review]

**Review by bmguiler — APPROVED** [review]
Looks good, your documentation is really clear and helpful to understand the flow of control

**Pull request merged**